install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -1300,8 +1300,8 @@ data:
   clustermesh-enable-endpoint-sync: {{ .Values.clustermesh.enableEndpointSliceSynchronization | quote }}
   clustermesh-enable-mcs-api: {{ .Values.clustermesh.enableMCSAPISupport | quote }}
 
-# Extra config allows adding arbitrary properties to the cilium config.
-# By putting it at the end of the ConfigMap, it's also possible to override existing properties.
+  # Extra config allows adding arbitrary properties to the cilium config.
+  # By putting it at the end of the ConfigMap, it's also possible to override existing properties.
```

Comment:
This solution doesn't work for me.

```
$ git log -1
commit f81c119d1950a9fea699644a337a11879cda0083 (HEAD -> alcidesmig/main)
Author: Alcides Mignoso <alcidesmig@gmail.com>
Date:   Mon Jul 15 16:41:30 2024 -0300

    fix(helm-chart): cilium-configmap extraConfig
    
    The `extraConfig` field description comments were incorrectly indented. When the option was used the generated ConfigMap was invalid, causing the installation to fail.
    
    Signed-off-by: Alcides Mignoso <alcidesmig@gmail.com>
$ helm install cilium ./install/kubernetes/cilium --namespace kube-system --version 1.15.3 --set extraConfig.enable-direct-routing="false"
Error: INSTALLATION FAILED: 1 error occurred:
        * ConfigMap in version "v1" cannot be handled as a ConfigMap: json: cannot unmarshal bool into Go struct field ConfigMap.data of type string


$ helm uninstall cilium -n kube-system
release "cilium" uninstalled
```

On the other hand, specifying the type as a string works:
```
$ helm install cilium ./install/kubernetes/cilium --namespace kube-system --version 1.15.3 --set-string extraConfig.enable-direct-routing="false"
NAME: cilium
LAST DEPLOYED: Mon Jul 15 18:51:22 2024
NAMESPACE: kube-system
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
You have successfully installed Cilium with Hubble.

Your release version is 1.17.0-dev.

For any further help, visit https://docs.cilium.io/en/v1.17/gettinghelp
```

Reply:
You are right, It was a mistake on my side. @aanm helped clarify here: https://github.com/cilium/cilium/issues/33829#issuecomment-2229530548